challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map(&:capitalize)
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  ['dog'] if input.include? 'dog'
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |word| word == 'dog' }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.length
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.select { |word| word.start_with?('c') or word.start_with?('d') }
```

Comment:
Interesting to see `or` being used, most people default to `||` ! Here's a quick read up on the differences and their use cases.